src/api/schema/gen.rs

```diff
@@ -0,0 +1,111 @@
+use std::fs;
+use vector::api::schema::build_schema;
+
+static INTROSPECTION_QUERY: &str = r#"
+query IntrospectionQuery {
+  __schema {
+    queryType {
+      name
+    }
+    mutationType {
+      name
+    }
+    subscriptionType {
+      name
+    }
+    types {
+      ...FullType
+    }
+    directives {
+      name
+      description
+      locations
+      args {
+        ...InputValue
+      }
+    }
+  }
+}
+fragment FullType on __Type {
+  kind
+  name
+  description
+  fields(includeDeprecated: true) {
+    name
+    description
+    args {
+      ...InputValue
+    }
+    type {
+      ...TypeRef
+    }
+    isDeprecated
+    deprecationReason
+  }
+  inputFields {
+    ...InputValue
+  }
+  interfaces {
+    ...TypeRef
+  }
+  enumValues(includeDeprecated: true) {
+    name
+    description
+    isDeprecated
+    deprecationReason
+  }
+  possibleTypes {
+    ...TypeRef
+  }
+}
+fragment InputValue on __InputValue {
+  name
+  description
+  type {
+    ...TypeRef
+  }
+  defaultValue
+}
+fragment TypeRef on __Type {
+  kind
+  name
+  ofType {
+    kind
+    name
+    ofType {
+      kind
+      name
+      ofType {
+        kind
+        name
+        ofType {
+          kind
+          name
+          ofType {
+            kind
+            name
+            ofType {
+              kind
+              name
+              ofType {
+                kind
+                name
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+"#;
+
+#[tokio::main]
+async fn main() {
+    let schema = build_schema().finish();
+    let res = schema.execute(INTROSPECTION_QUERY).await;
```

Comment:
This executes the introspection query against the live schema, outside of the Warp instantiation. This is interesting IMO, because it shows how we could bypass HTTP altogether and add a middle tier for marshalling requests via essentially any protocol. HTTP isn't a hard requirement!

Reply:
I wonder if this could be used by things like our wasm engine. 🤔

Reply:
[graphql-client](https://github.com/graphql-rust/graphql-client) is WASM-compatible; all queries, subscriptions, (de)serialisation of payloads etc should work.

I used [tokio-tungstenite](https://github.com/snapview/tokio-tungstenite) as the WebSocket client and [async-graphql](https://github.com/async-graphql/async-graphql) as the GraphQL server. I'm not sure if/how we'd intend to use these in wasm, but there's probably also the option of leveraging the native browser APIs for WebSockets.

Let me know if you have any specific use-cases in mind for leveraging GraphQL with WASM-- interesting!

Reply:
I definitely want to expose a `query(some_graphql_string)` via the hostcall API. :) I'll do that later though!